pkg/registry/chains/registryk8s/registry-k8s_test.go

```diff
@@ -400,7 +401,84 @@ func TestNSMGR_InterdomainUseCase(t *testing.T) {
 	_, err = nsc.Close(ctx, conn)
 	require.NoError(t, err)
 }
+func TestNSMGR_HealRegistry(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
 
+	const timeout = time.Second * 5
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	clientSet := fake.NewSimpleClientset()
+
+	domain := sandbox.NewBuilder(ctx, t).
+		SetNodesCount(1).
+		SetRegistrySupplier(supplyK8sRegistryWithClientSet(clientSet)).
+		SetNSMgrProxySupplier(nil).
+		SetRegistryProxySupplier(nil).
+		Build()
+
+	nsRegistryClient := domain.NewNSRegistryClient(ctx, sandbox.GenerateTestToken)
+
+	nsReg, err := nsRegistryClient.Register(ctx, defaultRegistryService(t.Name()))
+	require.NoError(t, err)
+
+	nseReg := defaultRegistryEndpoint(nsReg.Name)
+
+	counter := new(count.Server)
+	domain.Nodes[0].NewEndpoint(ctx, nseReg, sandbox.GenerateTestToken, counter)
+
+	request := defaultRequest(nsReg.Name)
+
+	nsc := domain.Nodes[0].NewClient(ctx, sandbox.GenerateTestToken)
+
+	conn, err := nsc.Request(ctx, request.Clone())
+	require.NoError(t, err)
+
+	// 1. Restart Registry
+	domain.Registry.Restart()
+
+	// 2. Check refresh
+	request.Connection = conn
+	_, err = nsc.Request(ctx, request.Clone())
+	require.NoError(t, err)
+
+	// 3. Check new client request
+	nsc = domain.Nodes[0].NewClient(ctx, sandbox.GenerateTestToken)
```

Comment:
Am I right in understanding that here we test an NSC restart, reconnect with same connection ID?

Reply:
As I can see by default it will generate a new connection id. See at https://github.com/networkservicemesh/sdk/blob/main/pkg/networkservice/chains/client/client.go#L43 -> https://github.com/networkservicemesh/sdk/blob/main/pkg/networkservice/chains/client/client.go#L55